homeassistant/components/cast/media_player.py

```diff
@@ -476,6 +478,19 @@ def media_seek(self, position):
 
     def play_media(self, media_type, media_id, **kwargs):
         """Play media from a URL."""
+        if media_type == CAST_DOMAIN:
+            try:
+                app_data = json.loads(media_id)
+            except json.JSONDecodeError:
+                _LOGGER.error("Invalid JSON in media_content_id")
+            else:
+                app_name = app_data.pop("app_name")
+                try:
+                    quick_play(self._chromecast, app_name, app_data)
+                    return
+                except NotImplementedError:
+                    _LOGGER.error("App %s not supported", app_name)
+
```

Comment:
We want to return here after the error log, I think.

Reply:
Hmm, I figured we shoild try regular media casting on failure.

But maybe you're right, it should just fail since the user did choose the `media_content_type: cast` option.